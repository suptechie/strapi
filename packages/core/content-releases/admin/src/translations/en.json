{
  "plugin.name": "Releases",
  "pages.Releases.title": "Releases",
  "pages.Releases.header-subtitle": "{number, plural, =0 {No releases} one {# release} other {# releases}}",
  "header.actions.add-release": "New Release",
  "header.actions.refresh": "Refresh",
  "header.actions.release": "Release",
  "header.actions.open-release-actions": "Release actions",
  "header.actions.edit": "Edit",
  "header.actions.delete": "Delete",
  "header.actions.created": "Created",
  "header.actions.created.description": "{number, plural, =0 {# days} one {# day} other {# days}} ago by {user}",
  "modal.release-created-notification-success": "Release created",
  "modal.add-release-title": "New Release",
  "modal.form.input.label.release-name": "Name",
  "modal.form.button.submit": "Continue",
  "pages.Details.header-subtitle": "{number, plural, =0 {No entries} one {# entry} other {# entries}}"
}
