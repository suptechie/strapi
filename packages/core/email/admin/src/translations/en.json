{
  "Settings.button.test-email": "Test email",
  "Settings.form.label.defaultFrom": "Default shipper email",
  "Settings.form.label.defaultReplyTo": "Default response  email",
  "Settings.form.label.provider": "Email provider",
  "Settings.form.label.testAddress": "Test delivery email address",
  "Settings.form.placeholder.defaultFrom": "ex: Strapi No-Reply <no-reply@strapi.io>",
  "Settings.form.placeholder.defaultReplyTo": "ex: Strapi <example@strapi.io>",
  "Settings.form.placeholder.testAddress": "ex: developer@example.com",
  "Settings.form.text.configuration": "The plugin is configured through the {file} file, checkout this {link} for the documentation.",
  "Settings.form.title.config": "Configuration",
  "Settings.form.title.test": "Send a test email",
  "Settings.notification.config.error": "Failed to retrieve the email config",
  "Settings.notification.test.error": "Failed to send a test mail to {to}",
  "Settings.notification.test.success": "Email test succeeded, check the {to} mailbox",
  "Settings.subTitle": "Test the settings for the email plugin",
  "Settings.title": "Email settings",
  "SettingsNav.link.settings": "Email settings",
  "SettingsNav.section-label": "Email plugin",
  "plugin.description.long": "Send emails.",
  "plugin.description.short": "Send emails."
}
