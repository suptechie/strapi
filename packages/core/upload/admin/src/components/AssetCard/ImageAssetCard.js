import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import {
  Card,
  CardAction,
  CardAsset,
  CardBadge,
  CardBody,
  CardCheckbox,
  CardContent,
  CardHeader,
  CardTitle,
  CardSubtitle,
} from '@strapi/parts/Card';
import { IconButton } from '@strapi/parts/IconButton';
import EditIcon from '@strapi/icons/EditIcon';
import { useIntl } from 'react-intl';
import { getTrad } from '../../utils';

const Extension = styled.span`
  text-transform: uppercase;
`;

export const ImageAssetCard = ({
  name,
  extension,
  height,
  width,
  thumbnail,
  selected,
  onSelect,
  onEdit,
  size,
}) => {
  const { formatMessage } = useIntl();

  // Prevents the browser from caching the URL for all sizes and allow react-query to make a smooth update
  // instead of a full refresh
  const optimizedCachingThumbnail =
    width && height ? `${thumbnail}?width=${width}&height=${height}` : thumbnail;

  return (
    <Card>
      <CardHeader>
        {onSelect && <CardCheckbox value={selected} onValueChange={onSelect} />}
        {onEdit && (
          <CardAction position="end">
            <IconButton
              label={formatMessage({ id: getTrad('control-card.edit'), defaultMessage: 'Edit' })}
              icon={<EditIcon />}
              onClick={onEdit}
            />
          </CardAction>
        )}
        <CardAsset src={optimizedCachingThumbnail} size={size} />
      </CardHeader>
      <CardBody>
        <CardContent>
          <CardTitle as="h2">{name}</CardTitle>
          <CardSubtitle>
            <Extension>{extension}</Extension>
            {height && width && `- ${height}âœ•${width}`}
          </CardSubtitle>
        </CardContent>
        <CardBadge>
          {formatMessage({ id: getTrad('settings.section.image.label'), defaultMessage: 'Image' })}
        </CardBadge>
      </CardBody>
    </Card>
  );
};

ImageAssetCard.defaultProps = {
  height: undefined,
  width: undefined,
  selected: false,
  onEdit: undefined,
  onSelect: undefined,
  size: 'M',
};

ImageAssetCard.propTypes = {
  extension: PropTypes.string.isRequired,
  height: PropTypes.number,
  name: PropTypes.string.isRequired,
  onEdit: PropTypes.func,
  onSelect: PropTypes.func,
  width: PropTypes.number,
  thumbnail: PropTypes.string.isRequired,
  selected: PropTypes.bool,
  size: PropTypes.oneOf(['S', 'M']),
};
