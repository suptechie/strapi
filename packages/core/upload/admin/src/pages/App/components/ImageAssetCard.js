import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import {
  Card,
  CardAction,
  CardAsset,
  CardBadge,
  CardBody,
  CardCheckbox,
  CardContent,
  CardHeader,
  CardTitle,
  CardSubtitle,
} from '@strapi/parts/Card';
import { IconButton } from '@strapi/parts/IconButton';
import EditIcon from '@strapi/icons/EditIcon';
import { useIntl } from 'react-intl';
import { getTrad } from '../../../utils';

const Extension = styled.span`
  text-transform: uppercase;
`;

export const ImageAssetCard = ({ name, extension, height, width, thumbnail }) => {
  const { formatMessage } = useIntl();

  return (
    <Card>
      <CardHeader>
        <CardCheckbox value />
        <CardAction position="end">
          <IconButton
            label={formatMessage({ id: getTrad('control-card.edit'), defaultMessage: 'Edit' })}
            icon={<EditIcon />}
          />
        </CardAction>
        <CardAsset src={`http://localhost:1337${thumbnail}`} />
      </CardHeader>
      <CardBody>
        <CardContent>
          <CardTitle as="h2">{name}</CardTitle>
          <CardSubtitle>
            <Extension>{extension.replace('.', '')}</Extension> - {height}âœ•{width}
          </CardSubtitle>
        </CardContent>
        <CardBadge>
          {formatMessage({ id: getTrad('settings.section.image.label'), defaultMessage: 'Image' })}
        </CardBadge>
      </CardBody>
    </Card>
  );
};

ImageAssetCard.propTypes = {
  extension: PropTypes.string.isRequired,
  height: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  width: PropTypes.number.isRequired,
  thumbnail: PropTypes.string.isRequired,
};
