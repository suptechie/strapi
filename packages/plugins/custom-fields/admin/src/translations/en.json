{
  "input.format": "HEX",
  "color-picker.label": "Color Picker",
  "color-picker.description": "Select any color" 
}
