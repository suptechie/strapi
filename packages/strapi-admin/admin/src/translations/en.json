{
  "app.components.ComingSoonPage.comingSoon": "Coming soon",
  "app.components.ComingSoonPage.featuresNotAvailable": "This feature is still under active development.",
  "app.components.HomePage.welcome": "Welcome on board!",
  "app.components.HomePage.description.part1": "We are happy to have you as one of our users!",
  "app.components.HomePage.description.part2": "You are now a member of our community which will help you building your dreamed app.",
  "app.components.HomePage.button": "Create your first content type",
  "app.components.HomePage.feedback": "Feel free to ask questions or give us feedback by using one of the support channels below.",
  "app.components.LeftMenuFooter.poweredBy": "Proudly powered by",
  "app.components.LeftMenuLinkContainer.configuration": "Configuration",
  "app.components.LeftMenuLinkContainer.general": "General",
  "app.components.LeftMenuLinkContainer.installNewPlugin": "Install new plugin",
  "app.components.LeftMenuLinkContainer.listPlugins": "List plugins",
  "app.components.LeftMenuLinkContainer.noPluginsInstalled": "No plugins installed yet",
  "app.components.LeftMenuLinkContainer.plugins": "Plugins",
  "app.components.NotFoundPage.description": "Not Found",
  "app.components.NotFoundPage.back": "Back to homepage",
  "components.AutoReloadBlocker.header": "Reload feature is required for this plugin.",
  "components.AutoReloadBlocker.description": "Open the following file and enable the feature.",
  "components.ProductionBlocker.header": "This plugin is only available in development.",
  "components.ProductionBlocker.description": "For safety we have to disable this plugin in other environments.",
  "components.popUpWarning.button.cancel": "Cancel",
  "components.popUpWarning.button.confirm": "Confirm",
  "components.popUpWarning.title": "Please confirm",
  "components.popUpWarning.message": "Are you sure you want to delete this?",
  "components.Input.error.validation.required": "This value input is required.",
  "components.Input.error.validation.regex": "The value not match the regex.",
  "components.Input.error.validation.max": "The value is too high.",
  "components.Input.error.validation.min": "The value is too low.",
  "components.Input.error.validation.maxLength": "The value is too long.",
  "components.Input.error.validation.minLength": "The value is too short.",
  "components.Input.error.contentTypeName.taken": "This name already exists",
  "components.Input.error.attribute.taken": "This field name already exists",
  "components.Input.error.attribute.key.taken": "This value already exists",
  "components.Input.error.attribute.sameKeyAndName": "Can't be equals",
  "components.Input.error.validation.minSupMax": "Can't be superior",
  "components.ErrorBoundary.title": "Something wen't wrong...",
  "components.ListRow.empty": "There is no data to be shown."
}
