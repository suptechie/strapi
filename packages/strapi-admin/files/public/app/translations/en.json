{
  "app.components.LeftMenuFooter.poweredBy": "",
  "app.components.NotFoundPage.description": "",
  "settings-manager.components.PluginHeaderTitle.description": "",
  "settings-manager.components.PluginHeaderTitle.title": ""
}