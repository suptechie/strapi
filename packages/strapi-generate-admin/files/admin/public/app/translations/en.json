{
  "app.components.LeftMenuFooter.poweredBy": "",
  "app.components.NotFoundPage.description": ""
}