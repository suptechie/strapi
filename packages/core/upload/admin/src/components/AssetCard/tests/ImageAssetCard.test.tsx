import { DesignSystemProvider } from '@strapi/design-system';
import { render as renderTL } from '@testing-library/react';

import en from '../../../translations/en.json';
import { ImageAssetCard } from '../ImageAssetCard';

jest.mock('../../../utils/index', () => ({
  ...jest.requireActual('../../../utils/index'),
  getTrad: (x: string) => x,
}));

jest.mock('react-intl', () => ({
  useIntl: () => ({ formatMessage: jest.fn(({ id }: { id: keyof typeof en }) => en[id]) }),
}));

describe('ImageAssetCard', () => {
  it('snapshots the component', () => {
    const { container } = renderTL(
      <DesignSystemProvider>
        <ImageAssetCard
          alt=""
          name="hello.png"
          extension="png"
          height={40}
          width={40}
          thumbnail="http://somewhere.com/hello.png"
          selected={false}
          onSelect={jest.fn()}
          onEdit={jest.fn()}
          isUrlSigned={false}
        />
      </DesignSystemProvider>
    );

    expect(container).toMatchSnapshot();
  });
});
