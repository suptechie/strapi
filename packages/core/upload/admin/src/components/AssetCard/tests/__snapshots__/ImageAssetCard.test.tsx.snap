// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageAssetCard snapshots the component 1`] = `
.c0 {
  height: 100%;
  background: #ffffff;
  border-radius: 4px;
  border-style: solid;
  border-width: 1px;
  border-color: #eaeaef;
  box-shadow: 0px 1px 4px rgba(33, 33, 52, 0.1);
}

.c2 {
  position: relative;
}

.c12 {
  border-radius: 4px;
  display: inline-flex;
  cursor: pointer;
}

.c17 {
  padding-block-start: 8px;
  padding-block-end: 8px;
  padding-inline-start: 12px;
  padding-inline-end: 12px;
}

.c20 {
  padding-block-start: 4px;
}

.c25 {
  padding-block-start: 4px;
  flex-grow: 1;
}

.c28 {
  padding-inline-start: 8px;
  padding-inline-end: 8px;
  min-width: 20px;
  background: #eaeaef;
}

.c3 {
  align-items: center;
  justify-content: center;
  flex-direction: row;
  display: flex;
}

.c5 {
  gap: 8px;
  align-items: center;
  flex-direction: row;
  display: flex;
}

.c18 {
  align-items: flex-start;
  flex-direction: row;
  display: flex;
}

.c26 {
  align-items: center;
  flex-direction: row;
  display: flex;
}

.c29 {
  align-items: center;
  justify-content: center;
  flex-direction: row;
  display: inline-flex;
}

.c21 {
  font-size: 1.4rem;
  line-height: 1.43;
  color: currentcolor;
}

.c22 {
  font-size: 1.2rem;
  line-height: 1.33;
  color: #32324d;
  font-weight: 600;
}

.c23 {
  font-size: 1.2rem;
  line-height: 1.33;
  color: #666687;
}

.c32 {
  font-weight: 600;
  font-size: 1.1rem;
  line-height: 1.45;
  text-transform: uppercase;
  color: #666687;
  line-height: 1rem;
}

.c14 {
  border: 0;
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c30 {
  border-radius: 4px;
  padding-block: 0.7rem;
}

.c13 {
  text-decoration: none;
  padding-block: 0.7rem;
  padding-inline: 0.7rem;
  border: 1px solid #dcdce4;
  background: #ffffff;
  color: #32324d;
  color: #8e8ea9;
}

.c13:hover {
  background-color: #f6f6f9;
  color: #666687;
}

.c13:active {
  background-color: #eaeaef;
}

.c13[aria-disabled='true'] {
  border: 1px solid #dcdce4;
  background: #eaeaef;
  color: #666687;
  cursor: default;
}

.c6 {
  position: absolute;
  top: 12px;
  left: 12px;
}

.c9 {
  position: absolute;
  top: 12px;
  right: 12px;
}

.c16 {
  margin: 0;
  padding: 0;
  max-height: 100%;
  max-width: 100%;
  object-fit: contain;
}

.c15 {
  display: flex;
  justify-content: center;
  height: 16.4rem;
  width: 100%;
  background: repeating-conic-gradient(#f6f6f9 0% 25%, transparent 0% 50%) 50%/20px 20px;
  border-top-left-radius: 4px;
  border-top-right-radius: 4px;
}

.c27 {
  margin-left: auto;
  flex-shrink: 0;
}

.c31 {
  margin-left: 4px;
}

.c7 {
  background: #ffffff;
  width: 2rem;
  height: 2rem;
  border-radius: 4px;
  border: 1px solid #c0c0cf;
  position: relative;
  z-index: 0;
  display: flex;
  justify-content: center;
  align-items: center;
  flex: 0 0 2rem;
}

.c7[data-state='checked'],
.c7[data-state='indeterminate'] {
  border: 1px solid #4945ff;
  background-color: #4945ff;
}

.c7[data-disabled] {
  background-color: #dcdce4;
}

.c7::before {
  content: '';
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  width: 100%;
  height: 100%;
  z-index: -1;
  min-width: 44px;
  min-height: 44px;
}

.c8 {
  width: 100%;
  height: 100%;
  cursor: pointer;
}

.c19 {
  word-break: break-all;
}

.c4 {
  border-bottom: 1px solid #eaeaef;
}

.c24 {
  text-transform: uppercase;
}

.c11 {
  opacity: 0;
}

.c11:focus-within {
  opacity: 1;
}

.c1 {
  cursor: pointer;
}

.c1:hover .c10 {
  opacity: 1;
}

@media (prefers-reduced-motion: no-preference) {
  .c13 {
    transition: background-color 120ms cubic-bezier(0.25, 0.46, 0.45, 0.94),color 120ms cubic-bezier(0.25, 0.46, 0.45, 0.94),border-color 200ms cubic-bezier(0.25, 0.46, 0.45, 0.94);
  }
}

<div>
  <article
    aria-labelledby=":r0:-title"
    class="c0 c1"
    role="button"
    tabindex="-1"
  >
    <div
      class="c2 c3 c4"
    >
      <div>
        <div
          class="c5 c6"
        >
          <button
            aria-checked="false"
            aria-labelledby=":r0:-title"
            class="c7"
            data-state="unchecked"
            role="checkbox"
            type="button"
            value="on"
          >
            <span
              class="c8"
              data-state="unchecked"
              style="pointer-events: auto; display: inline-flex;"
            />
          </button>
        </div>
      </div>
      <div
        class="c5 c9 c10 c11"
      >
        <button
          aria-disabled="false"
          class="c12 c3 c13"
          data-state="closed"
        >
          <svg
            aria-hidden="true"
            fill="currentColor"
            focusable="false"
            height="16"
            viewBox="0 0 32 32"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m28.414 9.171-5.585-5.586a2 2 0 0 0-2.829 0L4.586 19A1.98 1.98 0 0 0 4 20.414V26a2 2 0 0 0 2 2h5.586A1.98 1.98 0 0 0 13 27.414L28.414 12a2 2 0 0 0 0-2.829M24 13.585 18.414 8l3-3L27 10.585z"
            />
          </svg>
          <span
            class="c14"
          >
            Edit
          </span>
        </button>
      </div>
      <div
        class="c15"
      >
        <img
          alt=""
          aria-hidden="true"
          class="c16"
          src="http://somewhere.com/hello.png"
        />
      </div>
    </div>
    <div
      class="c17"
    >
      <div
        class="c18"
      >
        <div
          class="c19"
        >
          <div
            class="c20"
          >
            <h2
              class="c21"
            >
              <span
                class="c22"
                id=":r0:-title"
              >
                hello.png
              </span>
            </h2>
          </div>
          <div
            class="c23"
          >
            <span
              class="c24"
            >
              png
            </span>
             - 40âœ•40
          </div>
        </div>
        <div
          class="c25 c26"
        >
          <div
            class="c27"
          >
            <div
              class="c28 c29 c30 c31"
            >
              <span
                class="c32"
              >
                Image
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </article>
  <span
    class="c14"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </span>
</div>
`;
